docs/arch/adr-002.md

```diff
@@ -0,0 +1,38 @@
+# ADR 2: Code Generation Process
+
+## Context
+
+We will be generating a number of classes as per [ADR 1](adr-001.md).
+
+Generated classes should be generated during the build process and not committed to GitHub.
+This ensures we have reproducable builds and makes it clear that generated classes are not to be edited manually.
+
+We wish to publish articats to the central maven repository.
+This does not necessarily limit us to working with Maven, but it encourages us to stick it.
+
+Using a compiled language to run the code generation would require two passes of compilation
+(or another/project module) - one to compile the codegen scripts, another to run it.
```

Comment:
Maybe define that it has to do with annotation based processes and Immutable codegen. Would that be necessary?